src/dune_rules/install_rules.ml

```diff
@@ -43,6 +43,25 @@ module Package_paths = struct
     Path.Build.extend_basename (meta_file ctx pkg) ~suffix:".template"
 end
 
+let check_runtime_deps_relative_path local_path ~loc ~lib_info =
+  let lib_src_dir = Lib_info.src_dir lib_info in
+  match
+    Path.Local.descendant local_path ~of_:(Path.Build.local lib_src_dir)
+  with
+  | None ->
+    User_error.raise ~loc
+      [ Pp.textf
+          "Public library %s depends on a path defined in a parent directory. \
```

Comment:
Not sure if "in a parent directory" could be refined, because afaiu it doesn't necessarily need to be a parent directory.

Alternative: "Public library %s depends on an asset that is outside the library directory tree."

Reply:
Agree. Tweaked the message based on this suggestion.